modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFieldLocalServiceImpl.java

```diff
@@ -185,6 +185,21 @@ public ObjectField addOrUpdateCustomObjectField(
 				objectFieldSettings);
 		}
 
+		if (Objects.equals(businessType, "Relationship")) {
+			ObjectDefinition objectDefinition =
+				_objectDefinitionPersistence.findByPrimaryKey(
+					objectDefinitionId);
+
+			if (objectDefinition.isRootDescendantNode() &&
+				(Objects.equals(readOnly, "true") ||
+				 Objects.equals(readOnly, "conditional"))) {
```

Comment:
@larissacribeiro the ticket describes the expected behavior as: 

> we must validate that the user cannot update this property.

So I think it makes more sense to do:  
```
if (objectDefinition.isRootDescendantNode() && 
    !Objects.equals(existingObjectField.getReadOnly(), readOnly))
```

Can we do this?

Reply:
Yes, you're right! thanks for the review @carolmariaabb, just updated it with what you requested.